gcc/cp/contracts.cc

```diff
@@ -1717,6 +1891,69 @@ build_contract_violation (tree contract, contract_continuation cmode)
   return ctor;
 }
 
+static int
+get_contract_assertion_kind(tree contract)
+{
+  switch (TREE_CODE (contract))
+  {
+    case ASSERTION_STMT:	return CAK_ASSERT;
+    case PRECONDITION_STMT:	return CAK_PRE;
+    case POSTCONDITION_STMT:	return CAK_POST;
+    default:
+      error ("Invalid contract kind");
+  }
+
+  return CAK_INVALID;
+}
+
+static int
+get_evaluation_semantic(tree contract, contract_continuation cmode)
+{
+  contract_semantic semantic = get_contract_semantic (contract);
+
+  if (checked_contract_p(semantic))
+    {
+	  return CES_ENFORCE;
+    }
+  return CES_OBSERVE;
+}
+
+static tree
+build_contract_violation_P2900 (tree contract, contract_continuation cmode)
+{
+  int assertion_kind = get_contract_assertion_kind (contract);
+  int evaluation_semantic = get_evaluation_semantic (contract, cmode);
+  /* we hardcode CDM_PREDICATE_FALSE because that's all we support for now */
+  int detection_mode = CDM_PREDICATE_FALSE;
+
+  /* Must match the type layout in get_pseudo_contract_violation_type.  */
+  tree ctor = build_constructor_va
+    (init_list_type_node, 7,
+     NULL_TREE, build_int_cst (NULL_TREE, 0), 		// version
```

Comment:
all the integer constants here should have a defined type, right? (as set up in get_pseudo_contract_violation_type)

Reply:
changed to integer_type_node